services/audit-service/src/main/java/datawave/microservice/audit/auditors/file/FileAuditor.java

```
@@ -0,0 +1,220 @@
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URLEncoder;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * An implementation for {@link Auditor}, which writes URL encoded, JSON formatted audit messages to a file.
+ */
+public class FileAuditor implements Auditor {
+    
+    private final Logger log = LoggerFactory.getLogger(this.getClass());
+    
+    protected static final ObjectMapper mapper = new ObjectMapper();
+    
+    protected final SimpleDateFormat sdf;
+    protected final ReentrantLock writeLock = new ReentrantLock(true);
+    
+    protected long maxFileLenBytes;
+    protected long maxFileAgeMillis;
+    
+    protected FileSystem fileSystem;
+    protected Path basePath;
+    
+    protected Path currentFile = null;
+    protected Date creationDate = null;
+    
+    protected FileAuditor(Builder<?> builder) throws URISyntaxException, IOException {
+        this.maxFileLenBytes = builder.maxFileLenBytes;
+        this.maxFileAgeMillis = builder.maxFileAgeMillis;
+        
+        Configuration config = new Configuration();
+        config.set("fs.hdfs.impl", org.apache.hadoop.hdfs.DistributedFileSystem.class.getName());
+        config.set("fs.file.impl", org.apache.hadoop.fs.LocalFileSystem.class.getName());
+        
+        if (builder.configResources != null) {
+            for (String resource : builder.configResources)
+                config.addResource(new Path(resource));
+        }
+        
+        if (builder.fileUri != null)
+            fileSystem = FileSystem.get(new URI(builder.fileUri), config);
+        else
+            fileSystem = FileSystem.get(config);
```

this should never be allowed.  We always need a basePath (with protocol specified).  We need to fail if it is not specified.